src/types/profile-derived.js

```diff
@@ -124,8 +124,14 @@ export type CallNodeTable = {
  */
 export type CallNodeInfo = {
   callNodeTable: CallNodeTable,
-  // IndexIntoStackTable -> IndexIntoCallNodeTable
-  stackIndexToCallNodeIndex: Uint32Array,
+  // IndexIntoStackTable -> IndexIntoCallNodeTable | -1
+  // If this CallNodeInfo is inverted, this maps the uninverted stack to
+  // the inverted call node. For stacks which aren't used as self stacks, this
+  // is -1. In the non-inverted case, all entries are >= 0.
```

Comment:
I'm changing this to the following:

```
  // IndexIntoStackTable -> IndexIntoCallNodeTable | -1
  // If this CallNodeInfo is for the non-inverted tree, this maps the stack index
  // to its corresponding call node index, and all entries are >= 0.
  // If this CallNodeInfo is for the inverted tree, this maps the non-inverted
  // stack index to the inverted call node index. For example, the stack
  // A -> B -> C -> D is mapped to the inverted call node describing the
  // call path D <- C <- B <- A, i.e. the node with function A under the D root
  // of the inverted tree. Stacks which are only used as prefixes are not mapped
  // to an inverted call node; for those, the entry will be -1. In the example
  // above, if the stack node A -> B -> C only exists so that it can be the prefix
  // of the A -> B -> C -> D stack and no sample / marker / allocation has
  // A -> B -> C as its stack, then there is no need to have a call node
  // C <- B <- A in the inverted call node table.
```

This comment is temporary; after the "fast inverted tree" it will be gone.